stable/fluentd/templates/deployment.yaml

```diff
@@ -52,6 +52,15 @@ spec:
           - name: {{ $key | quote }}
             value: {{ $value | quote }}
           {{- end }}
+          {{- if .Values.envConfigMap }}
```

Comment:
What does this get us that we can get from the existing variables for setting environment variables?

Reply:
This sets variables from ConfigMaps. Existing env requires that variables be set in values, and for a chart that can be deployed in different environments, we need to be able to set variables from a config map in the environment.

Reply:
The change allows you to specify which config map and key to use to set the variable value.

Reply:
I understand what you are trying to do.
Why are you unable to use the existing functionality?

Reply:
Existing functionality only allows for simple key value pair (2 values in template), specified in values.yaml. This cannot be used to populate a value from config map because the format requires that the name of the variable, the config map to use, and the key to use (3 values in template) be provided. Additionally, the structure of the config is different.

I would be happy to use the existing functionality, but it does not provide what is needed.

Reply:
For reference, see how secrets are handled in stable/fluentd-elasticsearch:
From values.yaml
`secret:
# - name: ELASTICSEARCH_PASSWORD
#   secret_name: elasticsearch
#   secret_key: password`
From deployment.yaml
`        env:
        {{- if .Values.env }}
        {{- range $key, $value := .Values.env }}
        - name: {{ $key }}
          value: {{ $value | quote }}
        {{- end }}
        {{- end }}
        {{- if .Values.secret }}
        {{- range $key, $value := .Values.secret }}
        - name: {{ .name }}
          valueFrom:
            secretKeyRef:
              name: {{ $value.secret_name }}
              key: {{ $value.secret_key | quote }}
        {{- end }}
        {{- end }}`

Reply:
I'ts kiwigrid/fluentd-elasticsearch now ;-)
https://github.com/kiwigrid/helm-charts/tree/master/charts/fluentd-elasticsearch

Reply:
@rendhalver Can you clarify your objection to this change? From my perspective it adds valuable functionality that does not currently exist in the chart, and follows a pattern that already exists in other charts.